### DISCUSSION TITLE
FastAPI + Kerberos

### DISCUSSION THREAD

**Discussion by rmilkowski** [fastapi/fastapi#2743] [Questions] [answered]
Hi All,

How can I use FastAPI with Kerberos for user authentication?
I want users to authenticate with their kerberos ticket when connecting and FastAPI end points should be able to get krb user name which was authenticated. Apart from putting FastApi server behind proxy and injecting headers, is there a more "native" solution?

**Comment by gizmo84** [comment]
Hi @rmilkowski ,

did you find any solution on this?

**Reply by rmilkowski** [reply]
Nope.

**Comment by kaleming** [comment, marked as answer]
You can try using keycloak for authentication. There is support for kerberos.
